docs/adr/0001-add-generic-fetcher.md

```diff
@@ -0,0 +1,104 @@
+# Add generic fetcher
+
+## Context
+
+The main motivation for this change is to cover use cases of users that need to download arbitrary files that don't fit
+within an established package ecosystem cachi2 could potentially otherwise support. The target audience is users that
+want to use cachi2 to achieve hermetic builds and want an easy way to also include these arbitrary files, that cachi2
+will account for in the SBOM it produces.
+
+## Decision
+
+A new package manager for generic artifacts must be introduced.. This package manager utilizes a custom
+lockfile that is located in the input repository. Based on that lockfile, it will download files, save them into a requested
```

Comment:
```diff
- lockfile that is located in the input repository. Based on that lockfile, it will download files,
+ lockfile basing on which it will download files,
```

(since the file can be outside a repo now)

Reply:
Fixed.